### PR TITLE
Bluetooth: controller: Adds definitions for ISO PDUs

### PR DESCRIPTION
**Pull request by Thalley**
Adds the CIS, BIS and BIG Control PDU definitions.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

### CODE REVIEW

File: subsys/bluetooth/controller/ll_sw/pdu.h [hunk 1]
```diff
@@ -633,3 +633,69 @@ struct pdu_data {
 #endif /* CONFIG_BT_CTLR_PROFILE_ISR */
 	} __packed;
 } __packed;
+
+struct cis_pdu {
```
[the remark this answers is not in the document]
Comment (Thalley) [reply subsys/bluetooth/controller/ll_sw/pdu.h]: Completely agree; I wonder how I missed that. Thanks!

**Review by wopu-ot — APPROVED** [review]
LGTM

File: subsys/bluetooth/controller/ll_sw/pdu.h [hunk 2]
```diff
@@ -633,3 +633,69 @@ struct pdu_data {
...
+#elif __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+	uint8_t rfu:2;
+	uint8_t cstf:1;
+	uint8_t cssn:3;
+	uint8_t ll_id:2;
+#else
+#error "Unsupported endianness"
+#endif /* __BYTE_ORDER__ */
+	uint8_t length;
+	uint8_t payload[0];
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/pdu.h:676]:
```suggestion
	union {
		uint8_t payload[0];
		struct pdu_big_ctrl ctrl;
	} __packed;
```

File: subsys/bluetooth/controller/ll_sw/pdu.h [hunk 3]
```diff
@@ -633,3 +633,69 @@ struct pdu_data {
...
+} __packed;
+
+enum pdu_big_ctr_type {
+	PDU_BIG_CTR_TYPE_CHAN_MAP_IND = 0x00,
+	PDU_BIG_CTR_TYPE_TERM_IND = 0x01,
+};
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/pdu.h:682]:
```suggestion
enum pdu_big_ctrl_type {
	PDU_BIG_CTRL_TYPE_CHAN_MAP_IND = 0x00,
	PDU_BIG_CTRL_TYPE_TERM_IND = 0x01,
};
```

File: subsys/bluetooth/controller/ll_sw/pdu.h [hunk 4]
```diff
@@ -633,3 +633,69 @@ struct pdu_data {
...
+
+struct pdu_big_ctr_chan_map_ind {
+	uint8_t  chm[5];
+	uint16_t instant;
+} __packed;
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/pdu.h:687]:
```suggestion
struct pdu_big_ctrl_chan_map_ind {
	uint8_t  chm[5];
	uint16_t instant;
} __packed;
```

File: subsys/bluetooth/controller/ll_sw/pdu.h [hunk 5]
```diff
@@ -633,3 +633,69 @@ struct pdu_data {
...
+
+struct pdu_big_ctr_term_ind {
+	uint8_t  reason;
+	uint16_t instant;
+} __packed;
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/pdu.h:692]:
```suggestion
struct pdu_big_ctrl_term_ind {
	uint8_t  reason;
	uint16_t instant;
} __packed;
```

File: subsys/bluetooth/controller/ll_sw/pdu.h [hunk 6]
```diff
@@ -633,3 +633,69 @@ struct pdu_data {
...
+
+struct pdu_big_ctr {
+	uint8_t opcode;
+	union {
+		uint8_t ctr_data[0];
+		struct pdu_big_ctr_chan_map_ind chan_map_ind;
+		struct pdu_big_ctr_term_ind term_ind;
+	};
+} __packed;
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/pdu.h:701]:
```suggestion
struct pdu_big_ctrl {
	uint8_t opcode;
	union {
		uint8_t ctrl_data[0];
		struct pdu_big_ctrl_chan_map_ind chan_map_ind;
		struct pdu_big_ctrl_term_ind term_ind;
	} __packed;
} __packed;
```

**Review by cvinayak — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/controller/ll_sw/pdu.h [hunk 7]
```diff
@@ -633,3 +633,73 @@ struct pdu_data {
...
+
+
+struct pdu_big_ctrl {
+	uint8_t opcode;
+	union {
+		uint8_t ctrl_data[0];
+		struct pdu_big_ctrl_chan_map_ind chan_map_ind;
+		struct pdu_big_ctrl_term_ind term_ind;
+	};
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/pdu.h:683]:
```suggestion
	} __packed;
```

**Review by cvinayak — CHANGES_REQUESTED** [review]

**Review by cvinayak — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**